toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1693,29 +1695,56 @@ func selinuxRelabelFiles(systemConfig configuration.SystemConfig, installChroot
 	selinuxType := strings.TrimSpace(stdout)
 	fileContextPath := fmt.Sprintf(fileContextBasePath, selinuxType)
 
-	logger.Log.Debugf("Running setfiles to apply SELinux labels on mount points: %v", listOfMountsToLabel)
-	err = installChroot.UnsafeRun(func() error {
-		args := []string{"-m", "-v", fileContextPath}
-		args = append(args, listOfMountsToLabel...)
-
-		// We only want to print basic info, filter out the real output unless at trace level (Execute call handles that)
-		files := 0
-		lastFile := ""
-		onStdout := func(args ...interface{}) {
-			if len(args) > 0 {
-				files++
-				lastFile = fmt.Sprintf("%v", args)
+	targetRootPath := "/mnt/_bindmountroot"
+
+	for _, mountToLabel := range listOfMountsToLabel {
+		logger.Log.Debugf("Running setfiles to apply SELinux labels on mount points: %v", mountToLabel)
+
+		// Create a bind mount so that the filesystem's files can be labelled without interference from other mounts.
```

Comment:
Logic looks ok, but could you add a bit more context here? This looks a bit unnatural at first glance and this might help in the future.

Reply:
I've expanded the comment.